app/client/cli/utils.go

```diff
@@ -239,6 +249,45 @@ func attachChildPwdFlagToSubcommands() []cmdOption {
 	}}
 }
 
+func attachKeybaseFlagsToSubcommands() []cmdOption {
+	return []cmdOption{func(c *cobra.Command) {
+		c.Flags().StringVar(&kbTypeStrFromCLI, "keybase", "file", "keybase type used by the cmd, options are: file, vault")
+		c.Flags().StringVar(&keybaseVaultAddrFromCLI, "vault-addr", "", "Vault address used by the cmd. Defaults to https://127.0.0.1:8200 or VAULT_ADDR env var")
```

Comment:
So it's both, with the `POCKET_VAULT_ADDR` higher precedence. There is a caveat for the `POCKET_` prefix though. 

For `POCKET_VAULT_ADDR` to work, you need to have the `keybase.vault_addr` set in the config.json which is what I found when testing manually. I believe if we use their [`SetDefault` mechanism](https://pkg.go.dev/github.com/spf13/viper#readme-establishing-defaults) mechanism it _could_ work but I haven't confirmed this.

Hashicorp clients have standard env vars (similar to AWS) that they look for when initializing an SDK. `VAULT_ADDR` and `VAULT_TOKEN` are some of the more prominent ones.

Reply:
i think i found a workaround, stay tuned.

Reply:
> Hashicorp clients have standard env vars (similar to AWS) that they look for when initializing an SDK. VAULT_ADDR and VAULT_TOKEN are some of the more prominent ones.

Makes sense, let's keep it.

> For POCKET_KEYBASE_VAULT_ADDR to work, you need to have the keybase.vault_addr set in the config.json which is what I found when testing manually. I believe if we use their [SetDefault mechanism](https://pkg.go.dev/github.com/spf13/viper#readme-establishing-defaults) it could work without having it set in the config.json but I haven't confirmed this.

@deblasis actually ran into it in the past, and here's his fix: https://github.com/pokt-network/pocket/compare/issue/cli-viper-environment-vars-fix

@deblasis Why did we never merge this in? I figure @0xBigBoss can include it here.

Reply:
That was the first approach I tried last night, it didn't work well for the nested struct values from what I remember. The approach I took in this PR handles the nested config structs. You can pass map[string]any to SetDefault and Viper handles overriding nested struct keys automagically.

> Establishing Defaults
> A good configuration system will support default values. A default value is not required for a key, but it’s useful in the event that a key hasn't been set via config file, environment variable, remote configuration or flag.
> 
> Examples:
> 
> viper.SetDefault("ContentDir", "content")
> viper.SetDefault("LayoutDir", "layouts")
> viper.SetDefault("Taxonomies", map[string]string{"tag": "tags", "category": "categories"})